### PR TITLE
[IT-1531] Use sceptrelint validate stack tags

### PR DESCRIPTION
**Pull request by zaro0508**
Use the new precommit linter to validate that the assigned CostCenter
tags are valid.

depends on https://github.com/Sceptre/sceptrelint/pull/6

### CODE REVIEW

**Review by BrunoGrandePhD — APPROVED** [review]
Neat! I appreciate that these generic linting rules are being contributed to the official sceptrelint repo. 

To test this, I temporarily changed the rule to use @zaro0508's fork since `v1.1.0` hasn't been minted yet (as Khai pointed out as a dependency in his PR description). 

```yaml
-   repo: https://github.com/zaro0508/sceptrelint/
    rev: check_valid_tag_values
    [...]
```

Everything seems to work as expected!

```console
# Linting the scicomp repo as is
❯ pre-commit run --all-files check-stack-tag-values
[...]
check stack tag values...................................................Passed

# Linting after introducing an intentional typo in a cost center
❯ pre-commit run --all-files check-stack-tag-values
[...]
check stack tag values...................................................Failed
- hook id: check-stack-tag-values
- exit code: 1

- config/prod/htan-dcc-center-a.yaml: "HTAN-DFCI typo / 120100" value is not a valid CostCenter
```

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -18,9 +18,11 @@ repos:
...
-    rev: v1.0.1
+    rev: v1.1.0
     hooks:
     -    id: check-file-names
     -    id: check-stack-names
-    -    id: check-stack-tags
-         args: [--tag=CostCenter]
+    -    id: check-stack-tag-values
+         args: [--tag=CostCenter,
+                --file=https://raw.githubusercontent.com/Sage-Bionetworks/aws-infra/master/templates/tags/SageFinancialProgramCodes.json,
```
Comment (brucehoff) [inline .pre-commit-config.yaml:27]: Validation should either (1) disallow CostCenter=Other / 000001 or (2) allow  CostCenter=Other / 000001 but then also require that there is a CostCenterOther tag with a value other than Other / 000001.

**Review by brucehoff — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (zaro0508) [reply .pre-commit-config.yaml:27]: going with option 1.

**Review by brucehoff — APPROVED** [review]

**Comment by zaro0508** [comment]
thanks for testing it @BrunoGrandePhD

**Pull request merged**